app/src/main/java/com/simformsolutions/oneloop/ui/user/signup/signupphonevarification/SignupPhoneVerificationScreen.kt

```diff
@@ -0,0 +1,297 @@
+                singleLine = true,
+                keyboardOptions = KeyboardOptions.Default.copy(
+                    keyboardType = KeyboardType.Phone
+                ),
+                leadingIcon = {
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        AppIcon(
+                            modifier = Modifier.padding(end = dimensionResource(id = R.dimen.common_space_very_small)),
+                            painter = painterResource(
+                                id = R.drawable.icon_phone
+                            )
+                        )
+
+                        Row(
+                            modifier = Modifier
+                                .clickable {
+                                    selectCountryCode()
+                                },
+                            verticalAlignment = Alignment.CenterVertically
+                        ) {
+                            AppText(
+                                text = uiState.countryCode.ifEmpty { stringResource(R.string.select_country_code) }, // Display selected country code
+                                textColor = AppTheme.appColorScheme.textColor,
+                            )
+                            AppIcon(
+                                painter = painterResource(id = R.drawable.icon_dropbox), // Your dropdown icon
+                                tint = AppTheme.appColorScheme.textColor
+                            )
+
+                            Spacer(modifier = Modifier.width(dimensionResource(id = R.dimen.common_space_very_small)))
+                        }
+                    }
+                }
+            )
+
+            // Dialog to select country code
+            if (isDialogVisible) {
+                CountryCodeDialog(countryCodeList = uiState.countryCodeList,
+                    onCountryCodeSelect = { selectedCode ->
+                        updateCountryCode(selectedCode)
+                    },
+                    onDismiss = {
+                        updateCountryCode(uiState.countryCode)
+                    })
+            }
+
+            AppButton(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(
+                        top = dimensionResource(id = R.dimen.login_and_forgot_pass_buttons_top_padding)
+                    ),
+                contentPadding = PaddingValues(vertical = dimensionResource(id = R.dimen.common_space_medium)),
+                enabled = uiState.isValidInputOfPhoneNumberVerificationPage,
+                onClick = onPhoneNumberVerificationNextClicked
+            ) {
+                AppText(
+                    text = stringResource(R.string.btn_next),
+                    textColor = AppTheme.appColorScheme.white,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun CountryCodeDialog(
```

Comment:
It's not required to write dialog here, You can utilise `dialog` instead. It's similar to `composable` which is currently being used for showing screen. It will look something like below,

```kotlin
// SomeNavigation.kt

dialog<SelectCountry>(...) {
    SelectCountryRoute(...)
}
```